core/esmf-aspect-model-validator/src/main/java/org/eclipse/esmf/aspectmodel/shacl/constraint/NodeConstraint.java

```diff
@@ -14,20 +14,30 @@
 package org.eclipse.esmf.aspectmodel.shacl.constraint;
 
 import java.util.List;
+import java.util.Optional;
 
 import org.apache.jena.rdf.model.RDFNode;
+import org.apache.jena.rdf.model.Statement;
 import org.eclipse.esmf.aspectmodel.shacl.Shape;
+import org.eclipse.esmf.aspectmodel.shacl.path.Path;
 import org.eclipse.esmf.aspectmodel.shacl.violation.EvaluationContext;
 import org.eclipse.esmf.aspectmodel.shacl.violation.Violation;
 
 /**
  * Implements <a href="https://www.w3.org/TR/shacl/#NodeConstraintComponent">sh:node</a>
- * @param shape the node shape this sh:node refers to
+ * @param targetShape the node shape this sh:node refers to
  */
-public record NodeConstraint(Shape.Node shape) implements Constraint {
+public record NodeConstraint(Optional<Path> path, Shape.Node targetShape) implements Constraint {
```

Comment:
Should the Optional parameter be placed after the mandatory ones?

Reply:
Nice catch, you are right